### PR TITLE
Skip agent pod when stopping pods

### PR DESCRIPTION
**Pull request by jerryblakley**

### CODE REVIEW

File: pkg/agent/agent.go [hunk 1]
```diff
@@ -207,10 +208,14 @@ func (k *Klocksmith) process(stop <-chan struct{}) error {
 	glog.Infof("Deleting %d pods", len(pods))
 	deleteOptions := &v1meta.DeleteOptions{}
 	for _, pod := range pods {
-		glog.Infof("Terminating pod %q...", pod.Name)
-		if err := k.kc.CoreV1().Pods(pod.Namespace).Delete(pod.Name, deleteOptions); err != nil {
-			glog.Errorf("failed terminating pod %q: %v", pod.Name, err)
-			// Continue anyways, the reboot should terminate it
+		if strings.Contains(pod.Name, "update-agent") {
```
Comment (spheromak) [inline pkg/agent/agent.go]: hrm may want to do a label here  instead of a name match.  you can set a label that  says  drain-skip or something

Comment (winmillwill) [reply pkg/agent/agent.go]: Why not do this in `k.getPodsForDeletion` ?

**Comment by jburianek** [comment]
So @spheromak and I talked about this yesterday, and we're wondering if instead of taking all the pods on the node and filtering out the ones we don't want to kill, maybe it would make more sense to just label the pods we *do* want to kill and go from there. Since there will probably only be a few that we care about orderly shutdown of.

Jenkins was one example that popped into mind, since maybe we'd want it to save its job queues to disk before rebooting so it can pick up where it left off after the reboot.

**Comment by jburianek** [comment]
I think that still leaves open the problem of any daemonset pods we kill immediately respawning though, before the reboot has a chance to happen. To solve that I think we'd have to set a taint on the node, and then kill the daemonset pods.

**Comment by jburianek** [comment]
So maybe a process like this?
* Set some sort of `NoSchedule` taint on the node
* Kill any pods that we want to kill, possibly waiting for them to terminate
* Reboot the node
* After reboot, unset the `NoSchedule` taint on the node

A couple things to keep in mind about the above process:
1. The cos-update-agent will need a toleration set for the `NoSchedule` taint, otherwise the agent won't start back up after the reboot
2. None of our other daemonsets will start up after the reboot until that `NoSchedule` taint is removed

**Comment by jerryblakley** [comment]
Sounds complicated. Is it really necessary? 
Correct me where I'm wrong, but:
1. Our current approach is to leave all pods running until reboot, and we've not expressed concerns about this. If so, we could easily adjust the filter to filter all pods out so none are left to terminate.
2. Are there pods which we think benefit from a more graceful "delete pod" experience before the reboot, even if they manage to get recreated before reboot? If so, then we should just worry about the agent pod itself and do what this change attempts to do.

**Comment by spheromak** [comment]
I think for our purposes we could just invert this. 

Add a label to your workload if you want it to be drained.  Since we run almost all daemonsets it's likely we don't want to drain in most cases. 

 I would be ok with making drain an opt-in thing vs a default try to do it. The other consideration here is that the disruption budget is deployed to these clusters. Effectively a failing pod could stop a reboot. Possibly for now we just want to ignore this aspect of the operator and just move forward. We can add a cli option to enable drain or something like that ?

**Comment by jburianek** [comment]
Closing this, as we don't do any stopping of pods before telling systemd to reboot now.

**Pull request closed without merging**